src/base/OpenMCCellAverageProblem.C

```diff
@@ -469,6 +473,19 @@ OpenMCCellAverageProblem::OpenMCCellAverageProblem(const InputParameters & param
     const openmc::DAGUniverse * dag = dynamic_cast<const openmc::DAGUniverse *>(u);
     if (dag->uses_uwuw()) // TODO: test
       mooseError("The 'skinner' does not currently support the UWUW workflow.");
+
+    // The newly-generated DAGMC cells could be disjoint in space, in which case
+    // it is impossible for us to know with 100% certainty a priori how many materials
+    // we would need to create. It's also true that even if we had a single density
+    // bin, that we'd still need to create one material for every new temperature
+    // cell. So, we require the user to have map_density_by_cell to false if there
+    // is going to be any density feedback at all
+    if (_map_density_by_cell && _has_fluid_blocks)
+    {
+      _map_density_by_cell = false;
+      mooseWarning("When skinning by density, 'map_density_by_cell' is overridden to be false!\n"
+        "You can silence this warning by setting 'map_density_by_cell' to false.");
```

Comment:
Just wanted to make it abundantly clear to the user that densities are still incorporated into feedback here.

```suggestion
      mooseWarning("When skinning by density, material densities are fed back as part\n"
        "of the skinning process and 'map_density_by_cell' is overridden to be false!\n"
        "You can silence this warning by setting 'map_density_by_cell' to false.");
```

Reply:
> Material densities are sent back regardless of the `map_density_by_cell` setting...

This is what I was trying to make extra clear in the warning message. Has my addition muddied the water?

I'm just thinking that a new user still working their way through the docs might have a problem with their run, turn on this setting, get a warning and be confused.

Reply:
> Your addition isn't really explaining why the warning is being thrown - what you have written is true, but it's not why the warning is being thrown.

I don't disagree with that, but it does seem to provide some helpful context as to how the simulation is going to behave. Maybe it's getting in the way of the point though?

I'd prefer (i). I think displaying a warning makes perfect sense in this case -- just wanted to give the user some assurance that density is still getting back into the OpenMC model.

Reply:
If you feel this isn't a useful change, we can take or leave it. No big.

Reply:
To make sure I understand correctly, `elem` would be the equivalent of `map_density_by_cell` being `False` and `cell` is equivalent to `True`?

In the context to which that setting applies (mapping onto a CSG/DAMC geometry without skinning), the Boolean option makes more sense to me after reading the description of this setting more carefully in the documentation.
I wonder if it would be most accurate to say in the warning that this setting is ignored regardless of its value when a skinner is present.

Thanks for the thorough consideration of this!!

Reply:
That's correct.

`map_density_by_cell` is not ever ignored - the valid user options just depend on whether they have skinning or not.

- **No skinning (for CSG or DAGMC):** density can be sent from MOOSE to OpenMC by either (i) trying to set a unique density in every cell (`map_density_by_cell = true` == `map_density_by = cell`) or (ii) trying to set a unique density only on each unique material, which could technically be filling multiple cells (`map_density_by_cell = false` == `map_density_by = material`)
- **Skinning:** user MUST send density according to material (`map_density_by_cell = false` == `map_density_by = material`). I am just trying to save some headache by overriding their setting if they tried `map_density_by_cell = true`.

I'll defer to you on which you think is clearer - the boolean vs. the enumeration. What do you think?

Reply:
I guess I'm still a bit confused. Isn't the nature of the temperature and density feedback entirely determined by the properties of the skinner object if one is used?

(really feeling the need for a term other than skinner today hah...)

Reply:
Yeah this is kind of in the weeds. With skinning, it is possible for the same material to fill multiple fluid cells, if the skinner ends up making non-contiguous cells. i.e. in the figure below, there are four light blue cells, each filled with the same material. `map_density_by = material` is going to set the density of that _single_ material by averaging over the elements which map to all four cells together.

![Screen Shot 2023-04-30 at 1 07 23 PM](https://user-images.githubusercontent.com/17039662/235733299-841e15ed-530b-430a-97ee-a476047bbeaf.png)

Otherwise, with `map_density_by = cell`, Cardinal is going to throw an error because the user put the same material in multiple cells, preventing density feedback on a cell-wise level.

Reply:
And in the later case we're overriding that value when a skinner is present, right?

Reply:
> I just thought it'd be more user-friendly to do that automatically

Totally. I think I'd still describe this parameter as being ignored with the presence of a skinner. So if it's present in the input file I think a warning like "`map_density_by_cell` is present in the input but will be ignored. Density mapping will be governed by the model skinning parameters." 

Sorry for all the nitpicking. There are just a lot of capabilities now and I'm hoping to make things as clear as possible for the growing user base.

Reply:
Ok I like that - that's clearer than what I was trying to convey.

Thanks for the detailed review! You're absolutely right, the code is getting more and more complex and be clear in documentation is important.

Reply:
Ok, change has been implemented - let me know if I translated this convo correctly.

Reply:
Looks great!! 🙌🏻  Thanks again for the discussion